src/providers/oauth2/types.ts

```diff
@@ -1,11 +1,23 @@
 import { OAuth2ProviderTypeValues } from "../../schema/providers/oauth2/types.ts";
 
+const IdentityUrlHttpMethodTypes = {
+  POST: "POST",
+  GET: "GET",
+} as const;
+
+export type IdentityUrlHttpMethodTypeValues =
```

Comment:
Do we want users to do something like this?

```ts
  const Provider2 = DefineOAuth2Provider({
    provider_key: providerKey2,
    provider_type: Schema.providers.oauth2.CUSTOM,
    options: {
      "client_id": "123.456",
      "scope": ["scope_a", "scope_b"],
      "identity_config": {
        "url": "https://example.com",
        "account_identifier": "account_identifier_string",
        "headers": { "key1": "header_1", "key2": "header_2" },
        "body": { "param1": "body_1", "param2": "body_2" },
        "http_method_type": IdentityUrlHttpMethodTypes.POST,
      },
    },
  });
```

For http method values , it could be reasonable to ask the user to always pass the string value instead of a reference to an object we define (this seems to be the implementation of the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#supplying_request_options)) we could keep the `typescript` simple and define the following
```ts
IdentityUrlHttpMethodTypeValues = "GET" | "POST"
```

let me know what you think?